src/Features/LanguageServer/Protocol/Handler/RequestExecutionQueue.cs

```diff
@@ -79,6 +121,13 @@ public RequestExecutionQueue(ILspSolutionProvider solutionProvider)
             // Create a task completion source that will represent the processing of this request to the caller
             var completion = new TaskCompletionSource<TResponseType>();
```

Comment:
what's the threading invariants around this being called at the same time as Shutdown/Initialize?

Reply:
As above, StreamJsonRpc. I think moving the lifetime to the higher level might make this clearer, will see.

Reply:
I think changing the lifetime here makes this clearer. Let me know what you think